### PR TITLE
Add better token handling under hazardous condition

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##

Reach better handling at problematic points

### CODE REVIEW

File: engines/python/setup/djl_python/tests/test_test_model.py [hunk 1]
```diff
@@ -140,3 +147,56 @@ def test_with_env_chat(self):
...
+        serving_properties = {
+            "engine": "Python",
+            "rolling_batch": "auto",
+            "model_id": model_id
+        }
+        result = handler.inference_rolling_batch(
+            inputs, serving_properties=serving_properties)
+        for key, value in result.items():
+            final_dict = json.loads(value)
+            self.assertEqual(final_dict["details"]["finish_reason"], 'error')
```
Comment (rohithkrn) [inline engines/python/setup/djl_python/tests/test_test_model.py:179]: test is failing here

**Review by frankfliu — APPROVED** [review]

**Review by rohithkrn — APPROVED** [review]

**Comment by rohithkrn** [comment]
I was doing some tests on my side. With current container when streaming error occurs, the response will be
```
1714866779400: {"token": {"id": [338], "text": "is", "log_prob": -0.3300172984600067}}
{"token": {"id": [263], "text": " a", "log_prob": -1.0825505554676056}}
{"token": {"id": [2107], "text": " great", "log_prob": -2.0450351238250732}}
{"token": {"id": [1342], "text": " example", "log_prob": -0.7814879417419434}}
{"error":"","code":424}
```

After this PR, the response will be
```
1714867941864: {"token": {"id": [338], "text": "is", "log_prob": -0.3300172984600067}}
{"token": {"id": [2000], "text": " called", "log_prob": -2.8891883194446564}}
{"token": {"id": [1346], "text": " "", "log_prob": -2.0550572872161865}}
{"token": {"id": [2177], "text": "Par", "log_prob": -0.09167909622192383}}
{"token": {"id": [-1], "log_prob": -1}, "generated_text": "is called "Par"}
```

look at the last line. So that way this change can break backwards compatibility in terms how users interpret response.

File: engines/python/setup/djl_python/rolling_batch/rolling_batch.py [hunk 2]
```diff
@@ -407,19 +407,14 @@ def try_catch_handling(self, *args, **kwargs):
...
-                "code": 424,
-                "error": ""
-            }
-            results = []
-            for i in range(
-                    len(self.active_requests) + len(self.pending_requests)):
-                results.append(err)
+            for request in self.active_requests:
+                token = Token([-1], "", -1, None)
+                request.set_next_token(token,
```
Comment (rohithkrn) [inline engines/python/setup/djl_python/rolling_batch/rolling_batch.py:412]: Do you think we should also set error response code? https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_runtime_InvokeEndpointWithResponseStream.html#sagemaker-runtime_InvokeEndpointWithResponseStream-response-ModelStreamError

File: engines/python/setup/djl_python/rolling_batch/rolling_batch.py [hunk 3]
```diff
@@ -407,19 +410,14 @@ def try_catch_handling(self, *args, **kwargs):
...
+                                       last_token=True,
+                                       finish_reason="error")
```
Comment (rohithkrn) [inline engines/python/setup/djl_python/rolling_batch/rolling_batch.py:417]: could we also add `repr(exception)` to the finish reason? something like `finish_reason=f"error: repr(e)" to let the users know what the error is?

Comment (lanking520) [reply engines/python/setup/djl_python/rolling_batch/rolling_batch.py:417]: for some customers, they do not want to expose error log in stack traces, those info can be used by hacker to intercept the content inside

**Pull request merged**